src/main/java/edu/ie3/datamodel/io/naming/FileNamingStrategy.java

```diff
@@ -0,0 +1,248 @@
+   * @param timeSeries Time series to derive naming information from
+   * @return An optional sub directory path
+   */
+  public <T extends TimeSeries<E, V>, E extends TimeSeriesEntry<V>, V extends Value>
+      Optional<String> getDirectoryPath(T timeSeries) {
+    Optional<String> maybeDirectoryName = directoryHierarchy.getSubDirectory(timeSeries.getClass());
+    String directoryPath;
+    if (!maybeDirectoryName.isPresent()) {
+      logger.debug("Cannot determine directory name for time series '{}'.", timeSeries);
+      return Optional.empty();
+    } else {
+      /* Make sure, the directory path does not start or end with file separator and in between the separator is harmonized */
+      directoryPath =
+          IoUtil.harmonizeFileSeparator(
+              maybeDirectoryName
+                  .get()
+                  .replaceFirst("^" + IoUtil.FILE_SEPARATOR_REGEX, "")
+                  .replaceAll(IoUtil.FILE_SEPARATOR_REGEX + "$", ""));
+      return Optional.of(directoryPath);
+    }
+  }
+
+  public Pattern getIndividualTimeSeriesPattern() {
+    String subDirectory = directoryHierarchy.getSubDirectory(IndividualTimeSeries.class).orElse("");
+
+    if (subDirectory.isEmpty()) {
+      return entityNamingStrategy.getIndividualTimeSeriesPattern();
+    } else {
+      /* Build the pattern by joining the sub directory with the file name pattern, harmonizing file separators and
+       * finally escaping them */
+      String joined =
+          FilenameUtils.concat(
+              subDirectory, entityNamingStrategy.getIndividualTimeSeriesPattern().pattern());
+      String harmonized = IoUtil.harmonizeFileSeparator(joined);
+      String escaped = harmonized.replace("\\", "\\\\");
+
+      return Pattern.compile(escaped);
+    }
+  }
+
+  public Pattern getLoadProfileTimeSeriesPattern() {
+    String subDirectory = directoryHierarchy.getSubDirectory(LoadProfileInput.class).orElse("");
+
+    if (subDirectory.isEmpty()) {
+      return entityNamingStrategy.getLoadProfileTimeSeriesPattern();
+    } else {
+      /* Build the pattern by joining the sub directory with the file name pattern, harmonizing file separators and
+       * finally escaping them */
+      String joined =
+          FilenameUtils.concat(
+              subDirectory, entityNamingStrategy.getLoadProfileTimeSeriesPattern().pattern());
+      String harmonized = IoUtil.harmonizeFileSeparator(joined);
+      String escaped = harmonized.replace("\\", "\\\\");
+
+      return Pattern.compile(escaped);
+    }
+  }
+
+  /**
+   * Extracts meta information from a file name, of a time series. Here, a file name <u>without</u>
+   * leading path has to be provided
+   *
+   * @param fileName File name
+   * @return The meeting meta information
+   */
+  public FileNameMetaInformation extractTimeSeriesMetaInformation(String fileName) {
+    return entityNamingStrategy.extractTimeSeriesMetaInformation(fileName);
```

Comment:
Please move the implementations also to here. Shouldn't be part of entity naming, because it directly deals with something file related.

Reply:
done